### PR TITLE
Inu 267 generic error

### PR DESCRIPTION
**Pull request by vonipchenko**

### CODE REVIEW

File: CommonTypes/Source/GenericError/NetworkError.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    init(value: Value, requestURL: URL?) {
+        self.value = value
+        self.requestURL = requestURL
+        switch value {
+        case .invalidHTTPStatusCode(let status):
+            let codeDsc: String = HTTPURLResponse.localizedString(forStatusCode: status)
+            errorDescription = NSLocalizedString("Response status code was unacceptable:", comment: "") + " \(status) (\(codeDsc))"
+        case .apiError(_, _, let error):
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/NetworkError.swift]: name variables properly

File: CommonTypes/Source/GenericError/ErrorInterpreter.swift [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import Foundation
+
+protocol ErrorInterpreterProtocol {
+
+    init()
+
+    func generateDescription(_ error: Error) -> String
+
+    func isNotFoundError(_ error: GenericErrorProtocol) -> Bool
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/ErrorInterpreter.swift]: make them class

File: CommonTypes/Source/GenericError/ErrorInterpreter.swift [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import Foundation
+
+protocol ErrorInterpreterProtocol {
+
+    init()
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/ErrorInterpreter.swift]: remove this

File: CommonTypes/Source/GenericError/GenericError.swift [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    func isInternalServerError() -> Bool {
+      return interpreter.isInternalServerError(self)
+    }
+}
+
+// MARK: - CommonError PresentableError
+
+extension GenericCommonError {
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/GenericError.swift]: do not separate

File: CommonTypes/Source/GenericError/GenericError.swift [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+struct GenericError<Interpreter: ErrorInterpreterProtocol>: GenericErrorProtocol {
+
+    let value: Error
+    let errorDescription: String
+
+    let interpreter: Interpreter
+
+    static func unknownError() -> GenericError<Interpreter> {
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/GenericError.swift]: remove
`MyError(GenericCommonError.unknown)` is preferable

File: CommonTypes/Source/GenericError/GenericError.swift [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        return GenericError(value: GenericCommonError.unknown)
+    }
+
+    static func notAuthorizedError() -> GenericError<Interpreter> {
+        return GenericError(value: GenericCommonError.notAuthorized)
+    }
+
+    init(value: Error, errorDescription: String? = nil) {
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/GenericError.swift]: `_ value`

File: CommonTypes/Source/GenericError/GenericError.swift [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        let currentInterpreter = Interpreter()
+        self.interpreter = currentInterpreter
+        self.errorDescription = errorDescription ?? currentInterpreter.generateDescription(value)
```
Comment (SanakaBarabaka) [inline CommonTypes/Source/GenericError/GenericError.swift]: remove

File: HTTPClient/Source/HTTPClientError.swift [hunk 8]
```diff
@@ -4,6 +4,19 @@
 //
 
 import Foundation
+import Alamofire
+
+extension AFError: NetworkErrorConvertible {
+    func networkError() -> NetworkError {
+        if let invalidCode = responseCode {
```
Comment (SanakaBarabaka) [inline HTTPClient/Source/HTTPClientError.swift:11]: why no check for type?

**Review by SanakaBarabaka — CHANGES_REQUESTED** [review]

File: HTTPClient/Source/HTTPClientError.swift (hunk 8 above)
Comment (vonipchenko) [reply HTTPClient/Source/HTTPClientError.swift:11]: responseCode - /// The response code of a `.responseValidationFailed` error.
so i don't see any sense to unwrap type to get the same value

**Review by SanakaBarabaka — APPROVED** [review]

**Pull request merged**